daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SResult.scala

```diff
@@ -76,34 +76,25 @@ object SResult {
       callback: Option[ContractId] => Boolean,
   ) extends SResult
 
-  final case class SResultNeedLocalKeyVisible(
-      stakeholders: Set[Party],
-      committers: Set[Party],
-      // Callback.
-      // returns the next expression to evaluate.
-      // In case of failure the call back does not throw but set machine.ctrl to an SErrorDamlException
-      callback: SVisibleByKey => Unit,
-  ) extends SResult
-
-  sealed abstract class SVisibleByKey
-  object SVisibleByKey {
+  sealed abstract class SVisibility
```

Comment:
Maybe rename this to `SVisibleToStakeholders` or `SStakeholderAreReaders` (I’m sure you can come up with a less shitty name).

For contract ids it does not describe visibility properly and while we deprecate divulgence it’s still around for a while.

Reply:
Done, thanks!